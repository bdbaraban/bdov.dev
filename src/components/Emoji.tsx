import React, { ReactElement } from 'react';

/**
 * Emoji component prop types
 */
interface EmojiProps {
  label: string;
  symbol: string;
}

/**
 * Generic a11y-friendly Emoji component
 */
const Emoji = ({ label, symbol }: EmojiProps): ReactElement => (
  <span
    className="emoji"
    role="img"
    aria-label={label ? label : ''}
    aria-hidden={label ? 'false' : 'true'}
  >
    {symbol}
  </span>
);

export default Emoji;
