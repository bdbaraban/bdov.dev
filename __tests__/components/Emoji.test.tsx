import React from 'react';
import { render } from '@testing-library/react';
import Emoji from 'components/Emoji';

const props = {
  label: 'A dog, a faithful, beloved pet canine',
  symbol: '🐕',
  className: 'emoji'
};

test('sets aria-hidden to "true" when label prop is NOT passed', (): void => {
  const { getByRole } = render(<Emoji symbol={props.symbol} />);

  expect(getByRole('img').getAttribute('aria-label')).toBeNull();
  expect(getByRole('img').getAttribute('aria-hidden')).toEqual('true');
});

test('sets aria-hidden to undefined when label prop IS passed', (): void => {
  const { getByLabelText } = render(
    <Emoji label={props.label} symbol={props.symbol} />
  );

  expect(getByLabelText(props.label).getAttribute('aria-hidden')).toBeNull();
});

test('renders and matches snapshot', (): void => {
  const { container } = render(<Emoji {...props} />);

  expect(container.firstChild).toMatchSnapshot();
});
