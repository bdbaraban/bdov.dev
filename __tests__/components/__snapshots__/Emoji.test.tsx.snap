// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and matches snapshot 1`] = `
<span
  aria-label="A dog, a faithful, beloved pet canine"
  class="emoji"
  role="img"
>
  🐕
</span>
`;
